### ISSUE TITLE
STL files and board revision

### ISSUE THREAD

**Issue by FranTerrassa** [Osmagtor/8-to-4-scart-switch#1]
Hello, first of all, thank you for sharing this project, the design is great and the image quality is very good!! Could you share the stl, since I have made it as is with the same specifications and I don't see myself capable of making a box for this project.
Thanks and greetings.

**Comment by Osmagtor** [comment]
Sure. Although I'm thinking about making a few adjustments to the board (mostly using the back copper layer as well once I have access to better PCB printing machinery) and making a proper case for it in the forseeable future, for now I can share the STL files of the case that I'm using. However, I must warn you that it is far for perfect and will probably need some tinkering to suit your needs. 

My 3D printer isn't particularly large, so the design of the top and bottom shells had to be split in two halves. For faster prototyping, both top halves and bottom halves connect with their counterparts like a jigsaw puzzle. The board and the top shell are then supposed to slot into the bottom shell. That done, both shells can be screwed together after gluing or heating two nuts into the designated holes in the bottom shell.

Due to the low tolerance levels of the printer though, it was hard to get all pieces and the board to fit together nicely. Having gone through enough PLA already, it was decided to keep the prototype design and simply file off the different pieces until they fit. Strong plastic glue was then used to keep the pieces for each shell firmly stuck to one another. 

It is also adviseable to use some kind of central support for the board once slotted in. The pieces weren't filed off enough to support the board fully and it allowed the board to cave in and flex when plugging any SCART cables. I ended up using some leftover button caps for the job. Speaking of which, note that because of the difference in size between the switches of the sets of push-buttons and the switch of the standalone push button for the RGB on/off port, there are two different button cap sizes.

Finally, make sure that the holes for the RCA ports line up and are large enough. As you can probably tell by now, the design is fairly underwhelming, and since there was still room for improvement on the design of the board itself as well and the design might unsurprisingly not work for everyone, I had decided to keep it private.

In any case, I hope that it can aid you design a better revision or at least one that works for you. Here are the design files and some more reference pictures.

[Case Design Files.zip](https://github.com/user-attachments/files/17755364/Case.Design.Files.zip)

![PXL_20240719_164259212](https://github.com/user-attachments/assets/ccb88360-c779-4215-90cf-bc61e27c5b1c)

![PXL_20240719_164309404](https://github.com/user-attachments/assets/26a454e8-d997-4fa2-9452-cfb4c64b80b4)

![PXL_20240719_175637144](https://github.com/user-attachments/assets/de653d03-ff70-4264-ba00-753e56b9de33)

![PXL_20240719_175656939](https://github.com/user-attachments/assets/f079ebcb-c56f-45e0-825b-c1963ad44bfa)

![PXL_20240719_175735896](https://github.com/user-attachments/assets/e47f85e0-ff0b-47f9-bd8e-67431c432770)

![PXL_20240719_175741203](https://github.com/user-attachments/assets/93d6b23f-3d78-4a37-a9da-5495ab0b8251)

**Comment by Osmagtor** [comment]
Sure. Although I'm thinking about making a few adjustments to the board (mostly using the back copper layer as well once I have access to better PCB printing machinery) and making a proper case for it in the forseeable future, for now I can share the STL files of the case that I'm using. However, I must warn you that it is far for perfect and will probably need some tinkering to suit your needs. 

My 3D printer isn't particularly large, so the design of the top and bottom shells had to be split in two halves. For faster prototyping, both top halves and bottom halves connect with their counterparts like a jigsaw puzzle. The board and the top shell are then supposed to slot into the bottom shell. That done, both shells can be screwed together after gluing or heating two nuts into the designated holes in the bottom shell.

Due to the low tolerance levels of the printer though, it was hard to get all pieces and the board to fit together nicely. Having gone through enough PLA already, it was decided to keep the prototype design and simply file off the different pieces until they fit. Strong plastic glue was then used to keep the pieces for each shell firmly stuck to one another. 

It is also adviseable to use some kind of central support for the board once slotted in. The pieces weren't filed off enough to support the board fully and it allowed the board to cave in and flex when plugging any SCART cables. I ended up using some leftover button caps for the job. Speaking of which, note that because of the difference in size between the switches of the sets of push-buttons and the switch of the standalone push button for the RGB on/off port, there are two different button cap sizes.

Finally, make sure that the holes for the RCA ports line up and are large enough. As you can probably tell by now, the design is fairly underwhelming, and since there was still room for improvement on the design of the board itself as well and the design might unsurprisingly not work for everyone, I had decided to keep it private.

In any case, I'm really glad to see that you like this project! I hope that the STL files can aid you design a better case revision or at least one that works for you.

[Case Design Files.zip](https://github.com/user-attachments/files/17755364/Case.Design.Files.zip)

![PXL_20240719_164259212](https://github.com/user-attachments/assets/e9cbfb3a-23c2-47bf-9a12-534b08af591c)

![PXL_20240719_164309404](https://github.com/user-attachments/assets/994985d6-af46-4b99-8792-18648c310068)

![PXL_20240719_175637144](https://github.com/user-attachments/assets/45735586-de48-44c4-a7ac-a1e12092fa46)

![PXL_20240719_175656939](https://github.com/user-attachments/assets/0205cdfe-8ec8-42a6-8942-159f277fbcdb)

![PXL_20240719_175735896](https://github.com/user-attachments/assets/0f5b0379-e62b-4ab3-996f-e007bb377d5f)

![PXL_20240719_175741203](https://github.com/user-attachments/assets/b7f3e392-c8f4-4653-adf2-1ee7c393f06c)

**Comment by FranTerrassa** [comment]
wow!! thanks a lot, my printer is not that big either, I will try to make it as good as possible. I made changes to the back plate so I wouldn't have to solder the bridges. And it works like a charm, really, thank you very much.

**Comment by Osmagtor** [comment]
No worries, I'm glad to help out! I'll keep the issue open, so feel free to share your work once it's done.

**Comment by FranTerrassa** [comment]
I'm sending you the modifications made in Kicad, I changed the footprints (just footprints) on the face since they were placed on the wrong side, and I added the bridges on the side where there was nothing, that way you avoid bridging with cables. I'll upload something more when I'm done.
Regards

**Comment by FranTerrassa** [comment]
![image](https://github.com/user-attachments/assets/0af92ac0-2d95-4344-bb5c-6bb1e01a528e)

![image](https://github.com/user-attachments/assets/b5889281-dfa3-4d90-be82-a71b50203a36)

**Comment by FranTerrassa** [comment]
Hello, I'm sending you the latest files that I've retouched, I've deleted the previous link so as not to cause confusion, I've turned the switch around and moved it forward a bit (it was upside down, the button, according to the fingerprint, was facing inwards). And I'm sending you the STL files. I've also modified them, taking into account what you told me about the plate sinking when putting in the SCRT, which doesn't happen anymore with the modified design. For my part, I've left it the way I like it. It's a very good project considering what it costs to buy one that's already made and works properly, this design meets all requirements. Thank you very much again. I'm here for anything.
Regards.
P.S.: Sorry if it's not read correctly since I use Google Translate, I'm Spanish.

[kicad.zip](https://github.com/user-attachments/files/17800498/kicad.zip)
[STL.zip](https://github.com/user-attachments/files/17800500/STL.zip)

**Comment by Osmagtor** [comment]
Nice! It's looking very good. Your design is certainly much slimmer. I like it. 

As for the standalone switch, I don't remember having any issues with its footprint in KiCAD, but if it was going to be a problem in its current position, you did well by modifying it. I'll certainly give it a look in the next revision. I'm also glad to see that the spacing between the SCART port cutouts is right. It's something that took a lot of time to get right.

By the way, would you mind sharing some pics of the board itself if you have any? (And if you don't have to take apart the case again, of course). It's always cool to see what the board looks like. It should be far less messy without all the jumper wires.

P.S. No worries. I can understand you without any issues. I'm also Spanish myself, but for the sake of reaching as many people as possible, I prefer to stick to English.

**Comment by FranTerrassa** [comment]
![IMG-20241125-WA0002](https://github.com/user-attachments/assets/cdf51fca-39df-48db-90f7-9f6421e99bc1)
![IMG-20241125-WA0003](https://github.com/user-attachments/assets/f66c9d47-baae-497f-8a36-6ea27d840f6b)
![IMG-20241125-WA0004](https://github.com/user-attachments/assets/ef13cee6-2d97-4ac6-be05-379e4d0e2937)
![IMG-20241125-WA0005](https://github.com/user-attachments/assets/40042ac4-1c0f-4acb-ad4c-39f59b01c41a)

Hi, sorry for the delay, I'm busier than a Roman leg. There are 2 photos that show what I had to file down to make the box fit. I hope you don't mind the liberty I took in putting the logo, since it's for me and I'm not going to sell it or anything like that. I liked my logo there. For anything, you know. If you're interested in any plaques, I still have 4 left ;).
